ovos_dinkum_listener/service.py

```diff
@@ -681,7 +681,32 @@ def _save_stt(self, audio_bytes, stt_meta, save_path=None):
             stt_audio_dir = Path(f"{self.default_save_path}/utterances")
         stt_audio_dir.mkdir(parents=True, exist_ok=True)
 
-        filename = hash_sentence(stt_meta["transcription"])
+        listener = self.config.get("listener")
+
+        # What is a good default?
+        # Where does the documentation for this feature go?
+        default_template = "utterance_{date}_{uuid}"
+        default_template = "{hash}-{uuid}"
+        default_template = "{date:%Y-%m-%dT%H%M%S%z}-{uuid}"
+        default_template = "{hash}-{date:%Y-%m-%dT%H%M%S%z}-{uuid}"
+
+        filename_template = listener.get("filename_template", default_template)
+        formatter = _TemplateFilenameFormatter()
+
+        @formatter.register('hash')
+        def transcription_hash():
+            # Build a hash of the transcription
+            try:
+                # handles legacy API
+                return hash_sentence(stt_meta.get('transcription'))
+            except KeyError:
+                # handles new API
+                tests = [t[0] for t in stt_meta.get('transcriptions')]
+                concat_text = '_'.join(tests)
+                return hash_sentence(concat_text)
```

Comment:
why not use transcriptions[0] instead of concat? i think that would be better

Reply:
Is there a chance that `len(transcriptions) == 0`? I could special case that to return null